### PR TITLE
Finish yaml refactor

### PR DESCRIPTION
**Pull request by spxtr**
This turned out to be pretty tricky, the obvious solutions didn't work for various reasons. Rather than trying to be clever, I did this instead: All upgrade jobs now `export` lots of stuff at the beginning and then call the runner for their branch (`e2e-runner.sh` for master, `e2e.sh` for previous branches). I am fairly sure that this is okay, since in `e2e.sh` on previous branches the variables just get overridden.

This makes the code much simpler than it would have been otherwise, and also simplifies cutting a new release branch.

Also, I've modified PR Jenkins to make it okay to remove `e2e.sh`. I'll want to get that in source control soon.

ref #20997

### CODE REVIEW

File: hack/jenkins/job-configs/kubernetes-upgrades.yaml [hunk 1]
```diff
@@ -109,72 +117,120 @@
...
+                export GINKGO_TEST_ARGS="--ginkgo.focus=Kubectl"
         - 'kubernetes-upgrade-{provider}-{version-old}-{version-new}-{step}':
             step: 'step3-upgrade-master'
-            branch: '{branch-new}'
             description: 'Upgrade the master from {version-old} to {version-new}.'
+            job-env: |
+                export E2E_OPT="--check_version_skew=false"
+                export E2E_UP="false"
+                export E2E_DOWN="false"
+                export GINKGO_TEST_ARGS="--ginkgo.focus=\[Feature:Upgrade\].*upgrade-master"
```
Comment (ihmccreery) [inline hack/jenkins/job-configs/kubernetes-upgrades.yaml]: This isn't going to work for v1.1; see https://github.com/kubernetes/kubernetes/blob/release-1.1/hack/jenkins/e2e.sh#L168

File: hack/jenkins/job-configs/kubernetes-upgrades.yaml [hunk 2]
```diff
@@ -109,72 +117,120 @@
...
+                export E2E_DOWN="false"
         - 'kubernetes-upgrade-{provider}-{version-old}-{version-new}-{step}':
             step: 'step5-upgrade-cluster'
-            branch: '{branch-new}'
             description: 'Upgrade the nodes from {version-old} to {version-new}.'
+            job-env: |
+                export E2E_OPT="--check_version_skew=false"
+                export E2E_UP="false"
+                export E2E_DOWN="false"
+                export GINKGO_TEST_ARGS="--ginkgo.focus=\[Feature:Upgrade\].*upgrade-cluster"
```
Comment (ihmccreery) [inline hack/jenkins/job-configs/kubernetes-upgrades.yaml]: Nor this.

File: hack/jenkins/job-configs/kubernetes-upgrades.yaml (hunk 1 above)
Comment (ihmccreery) [inline hack/jenkins/job-configs/kubernetes-upgrades.yaml]: Oh, I guess they get overwritten in v1.1's `e2e.sh`.  I'm wondering what happens when v1.3 changes; do we have to go back and change v1.2 as well?  It's not the worst thing in the world, but it's not great either.  It just scares me a little that we're tightly coupling logic across multiple release branches.

File: hack/jenkins/e2e.sh [hunk 3]
```diff
@@ -1,230 +0,0 @@
...
-# Define environment variables based on the Jenkins project name.
-# NOTE: Not all jobs are defined here. The hack/jenkins/e2e.sh in master and
-# release branches defines relevant jobs for that particular version of
-# Kubernetes.
-case ${JOB_NAME} in
-
-  # PR builder
-
-  # Runs a subset of tests on GCE in parallel. Run against all pending PRs.
-  kubernetes-pull-build-test-e2e-gce)
```
Comment (ihmccreery) [inline hack/jenkins/e2e.sh]: What happened to the PR builder?

Comment (spxtr) [inline hack/jenkins/e2e.sh]: I already made the change on PR builder. I need to get that Jenkins instance into source control soon.

File: hack/jenkins/job-configs/kubernetes-upgrades.yaml (hunk 1 above)
Comment (spxtr) [inline hack/jenkins/job-configs/kubernetes-upgrades.yaml]: The hope is that the labels won't change from now on.

File: hack/jenkins/job-configs/kubernetes-upgrades.yaml [hunk 4]
```diff
@@ -56,7 +56,11 @@
         daysToKeep: 7
     builders:
         - shell: |
-            curl -fsS --retry 3  "https://raw.githubusercontent.com/kubernetes/kubernetes/{branch}/hack/jenkins/e2e.sh" | bash -
+            {provider-env}
+            {project-env}
```
Comment (ihmccreery) [inline hack/jenkins/job-configs/kubernetes-upgrades.yaml]: `project-env` and `job-env` are pretty ambiguous.  Add a comment?  (Project is per-upgrade-flow, job is per-step, right?)

File: hack/jenkins/job-configs/kubernetes-upgrades.yaml (hunk 1 above)
Comment (ihmccreery) [inline hack/jenkins/job-configs/kubernetes-upgrades.yaml]: At least add a comment that these should work across all versions we're upgrade testing at any given time.

File: hack/jenkins/job-configs/kubernetes-upgrades.yaml [hunk 5]
```diff
@@ -109,72 +117,120 @@
...
+                export E2E_OPT="--check_version_skew=false"
+                export E2E_UP="false"
 
 - project:
     name: 'upgrade-gke'
+    provider-env: |
+        {gke-provider-env}
+        export JENKINS_FORCE_GET_TARS="y"
+        export FAIL_ON_GCP_RESOURCE_LEAK="false"
+    project-env: ''
```
Comment (ihmccreery) [inline hack/jenkins/job-configs/kubernetes-upgrades.yaml]: This is confusing to me; I assume it's for defaulting, but maybe just have no default, and stick an empty `project-env` into the pre-v1.2 projects?

Comment (ihmccreery) [inline hack/jenkins/job-configs/kubernetes-upgrades.yaml]: With a comment.

File: hack/jenkins/job-configs/kubernetes-upgrades.yaml [hunk 6]
```diff
@@ -109,72 +117,120 @@
...
+            runner-new: curl -fsS --retry 3  "https://raw.githubusercontent.com/kubernetes/kubernetes/release-1.1/hack/jenkins/e2e-runner.sh" | bash -
 
 - project:
     name: 'upgrade-gce'
+    provider-env: |
+        {gce-provider-env}
+        export NUM_NODES=5
+        export JENKINS_FORCE_GET_TARS="y"
+        export FAIL_ON_GCP_RESOURCE_LEAK="false"
+    project-env: ''
```
Comment (ihmccreery) [inline hack/jenkins/job-configs/kubernetes-upgrades.yaml]: Same as above.

**Comment by ihmccreery** [comment]
Okay, this mostly looks good; I'd like a lot more commentary, because this stuff is super confusing.

And, where did the PR builder go?

**Comment by ihmccreery** [comment]
Fixes #20997.

I RTFM'ed, and it looks like you've got the PR builder under control.

You also said the way things are would make cutting a new branch simpler.  We'll see soon enough!

Thanks for doing all this.

File: hack/jenkins/job-configs/kubernetes-upgrades.yaml (hunk 4 above)
Comment (spxtr) [inline hack/jenkins/job-configs/kubernetes-upgrades.yaml]: That's right, done.

File: hack/jenkins/job-configs/kubernetes-upgrades.yaml (hunk 5 above)
Comment (spxtr) [inline hack/jenkins/job-configs/kubernetes-upgrades.yaml]: Done.

**Comment by spxtr** [comment]
Added a few comments and removed the default project-env.

File: hack/jenkins/job-configs/kubernetes-upgrades.yaml (hunk 1 above)
Comment (ihmccreery) [inline hack/jenkins/job-configs/kubernetes-upgrades.yaml]: > The hope is that the labels won't change from now on.

I think that's an unreasonable hope, but we can cross that bridge when we come to it.

**Comment by ihmccreery** [comment]
LGTM.  Manual merge?  cc @k8s-oncall

**Comment by spxtr** [comment]
Yes. This shouldn't affect e2e tests at all.

**Pull request merged**